cmd/vulkan_sample/main.cpp

```diff
@@ -0,0 +1,1771 @@
+  VK_KHR_WIN32_SURFACE_EXTENSION_NAME
+};
+
+void write_error(HANDLE handle, const char* message) {
+  DWORD written = 0;
+  WriteConsole(handle, message, static_cast<DWORD>(strlen(message)), &written,
+               nullptr);
+}
+
+HMODULE LoadVulkan() {
+  return LoadLibrary("vulkan-1.dll");
+}
+
+void* GetLibraryProcAddr(HMODULE library, const char* function_name) {
+  return reinterpret_cast<void*>(GetProcAddress(library, function_name));
+}
+
+// Create Win32 window
+bool CreateNativeWindow() {
+  kOutHandle = GetStdHandle(STD_OUTPUT_HANDLE);
+  if (kOutHandle == INVALID_HANDLE_VALUE) {
+    AllocConsole();
+    kOutHandle = GetStdHandle(STD_OUTPUT_HANDLE);
+    if (kOutHandle == INVALID_HANDLE_VALUE) {
+      return false;
+    }
+  }
+
+  WNDCLASSEX window_class;
+  window_class.cbSize = sizeof(WNDCLASSEX);
+  window_class.style = CS_HREDRAW | CS_VREDRAW;
+  window_class.lpfnWndProc = &DefWindowProc;
+  window_class.cbClsExtra = 0;
+  window_class.cbWndExtra = 0;
+  window_class.hInstance = GetModuleHandle(NULL);
+  window_class.hIcon = NULL;
+  window_class.hCursor = NULL;
+  window_class.hbrBackground = NULL;
+  window_class.lpszMenuName = NULL;
+  window_class.lpszClassName = "Sample application";
+  window_class.hIconSm = NULL;
+  if (!RegisterClassEx(&window_class)) {
+    DWORD num_written = 0;
+    write_error(kOutHandle, "Could not register class");
+    return false;
+  }
+  RECT rect = {0, 0, LONG(1024), LONG(768)};
+
+  AdjustWindowRect(&rect, WS_OVERLAPPEDWINDOW, FALSE);
+
+  kNativeWindowHandle = CreateWindowEx(
+      0, "Sample application", "", WS_OVERLAPPEDWINDOW, CW_USEDEFAULT,
+      CW_USEDEFAULT, rect.right - rect.left, rect.bottom - rect.top, 0, 0,
+      GetModuleHandle(NULL), NULL);
+
+  if (!kNativeWindowHandle) {
+    write_error(kOutHandle, "Could not create window");
+    return false;
+  }
+
+  kNativeHInstance = reinterpret_cast<HINSTANCE>(
+      GetWindowLongPtr(kNativeWindowHandle, GWLP_HINSTANCE));
+  ShowWindow(kNativeWindowHandle, SW_SHOW);
+  return true;
+}
+
+void ProcessNativeWindowEvents() {
+  MSG msg;
+  while (PeekMessage(&msg, kNativeWindowHandle, 0, 0, PM_REMOVE)) {
+    TranslateMessage(&msg);
+    DispatchMessage(&msg);
+  }
+}
+
+#endif
+
+uint32_t inline GetMemoryIndex(const VkPhysicalDeviceMemoryProperties& properties,
+                               uint32_t required_index_bits,
+                               VkMemoryPropertyFlags required_property_flags) {
+  uint32_t memory_index = 0;
+  for (; memory_index < properties.memoryTypeCount; ++memory_index) {
+    if (!(required_index_bits & (1 << memory_index))) {
+      continue;
+    }
+
+    if ((properties.memoryTypes[memory_index].propertyFlags &
+         required_property_flags) != required_property_flags) {
+      continue;
+    }
+    break;
+  }
+  if (memory_index ==  properties.memoryTypeCount) {
+    return static_cast<uint32_t>(-1);
+  }
+  return memory_index;
+}
+
+
+int main(int argc, const char** argv) {
```

Comment:
I would prefer to split this up to functions but if it is ok for everyone else, I am good too.